pkg/kapp/resources/resources.go

```diff
@@ -52,6 +52,7 @@ type ResourcesImpl struct {
 	resourceTypes             ResourceTypes
 	coreClient                kubernetes.Interface
 	dynamicClient             dynamic.Interface
+	mutedDynamicClient        dynamic.Interface
```

Comment:
Are we planning to use mutex so that we don't use both the clients at the same time? If so, is there any way we can avoid making changes in all the places they are used?

Reply:
Had a discussion about this, but seems like we won't need the mutex unless the clients are used in parallel, which is not the case right now. I tried figuring out a way with lesser files changed but since we instantiate clients in the `reps_factory.go` but use the instances that are stored in the struct in `resources.go` while API calls are made adding a common wrapper might be a bitt complicated. Certainly worth digging into but might add unnecessary complexity